### PR TITLE
fix: migrate UI project access from blueprint to mantine

### PR DESCRIPTION
**Pull request by hrutik7**
Relates to: #5405

### CODE REVIEW

**Comment by IrakliJani** [comment]
hi @hrutik7 is this PR ready to be reviewed?

**Comment by hrutik7** [comment]
hey @IrakliJani not yet . there is something that need to be checked
I'll do it today

**Comment by IrakliJani** [comment]
cool, looking forward to it @hrutik7

File: packages/frontend/src/styles/blueprint-core.css [hunk 1]
```diff
@@ -8986,12 +8986,7 @@ label.bp4-label:not(.bp4-inline) .bp4-popover-target {
     right: 12px;
     top: 12px;
 }
-.bp4-html-select.bp4-fill,
-.bp4-html-select.bp4-fill select,
-.bp4-select.bp4-fill,
-.bp4-select.bp4-fill select {
-    width: 100%;
-}
```
Comment (joaoviana) [inline packages/frontend/src/styles/blueprint-core.css:8994]: could we revert this change please? I'm unsure if we're using Blueprint `select`s in other places, so it's less risky to keep until we fully migrate 👍

File: packages/frontend/src/components/ProjectAccess/ProjectAccessCreation/index.tsx [hunk 2]
```diff
@@ -227,17 +231,29 @@ const ProjectAccessCreation: FC<ProjectAccessCreationProps> = ({
...
+                        radius="xs"
+                        sx={{
+                            marginTop: '20px',
+                            marginLeft: '7px',
+                            backgroundColor: '#2D72D2',
+                            height: '30px',
+                            width: '96px',
+                            fontFamily: 'sans-serif',
+                            fontWeight: 'lighter',
+                        }}
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectAccess/ProjectAccessCreation/index.tsx:247]: same here (regarding custom styling). We should try to keep this as _untouched_ as possible, so for example:
* if you want it with less padding, perhaps use `compact`? 
* no custom fonts, background color, margins
* also, try to use the Mantine props for box-model-related styles: `m`/`mx`/`h`/`w`, etc

File: packages/frontend/src/components/ProjectAccess/ProjectAccessCreation/index.tsx [hunk 3]
```diff
@@ -135,7 +133,13 @@ const ProjectAccessCreation: FC<ProjectAccessCreationProps> = ({
     };
 
     return (
-        <Panel>
+        <Box
+            sx={{
+                height: '100%',
+                display: 'flex',
+                flexDirection: 'column',
+            }}
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectAccess/ProjectAccessCreation/index.tsx:141]: Not sure if this is needed 🤔 what would just `Box` look like?

File: packages/frontend/src/components/ProjectAccess/ProjectAccess.tsx [hunk 4]
```diff
@@ -57,50 +63,76 @@ const UserListItem: FC<{
...
+                            arrowSize={10}
+                            label={`This user inherits the organization role: ${relevantOrgRole}`}
+                            sx={{ padding: '10px', backgroundColor: '#57606a' }}
                         >
-                            <RelevantOrgRoleIcon
-                                icon="warning-sign"
-                                intent={Intent.WARNING}
+                            <MantineIcon
+                                icon={IconAlertTriangleFilled}
+                                style={{ marginRight: '10', color: '#935610' }}
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectAccess/ProjectAccess.tsx:83]: you can pass `color` prop to this icon 👍

File: packages/frontend/src/components/ProjectAccess/ProjectAccess.styles.ts [hunk 5]
```diff
@@ -26,19 +25,42 @@ export const UserName = styled.b`
     margin-right: 0.625em;
 `;
 
-export const UserEmail = styled(Tag)`
-    width: fit-content;
-    margin-top: 0.3em;
+export const ButtonGroup = styled.div`
+    display: flex;
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectAccess/ProjectAccess.styles.ts:29]: I would avoid improving/keeping Styled components here, could we do the same with moving these styles + components from `ProjectAccess.styles.ts` to `ProjectAccess.tsx` without using Styled components? Apologies if this is not clear

File: packages/frontend/src/components/ProjectAccess/ProjectAccess.tsx [hunk 6]
```diff
@@ -57,50 +63,76 @@ const UserListItem: FC<{
...
                                         .value as ProjectMemberRole;
                                     onUpdate(newRole);
                                 }}
+                                variant="filled"
+                                sx={{
+                                    marginRight: '0.5em',
+                                    boxSizing: 'border-box',
+                                    height: '30px',
+                                    fontSize: '14px',
+                                }}
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectAccess/ProjectAccess.tsx:109]: What about using Mantine's `Select` instead? would be cool 😎

File: packages/frontend/src/components/ProjectAccess/ProjectAccess.tsx [hunk 7]
```diff
@@ -57,50 +63,76 @@ const UserListItem: FC<{
...
                                 value={role}
                             />
                         ) : (
-                            <Tooltip2
-                                content={roleTooltip ? roleTooltip : undefined}
+                            <Tooltip
+                                position="left"
+                                color="dark"
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectAccess/ProjectAccess.tsx:115]: not sure if we need this?

File: packages/frontend/src/components/ProjectAccess/ProjectAccess.tsx [hunk 8]
```diff
@@ -213,8 +244,13 @@ const ProjectAccess: FC<ProjectAccessProps> = ({ projectUuid }) => {
         }),
     );
 
-    if (isProjectAccessLoading || isOrganizationUsersLoading) {
-        return <NonIdealState title="Loading..." icon={<Spinner />} />;
+    if (isProjectAccessLoading || isProjectAccessLoading) {
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectAccess/ProjectAccess.tsx:247]: this removes the ability to check if `isOrganizationUsersLoading` is true. Can we bring it back?

**Review by joaoviana — CHANGES_REQUESTED** [review]
Thanks for your contribution @hrutik7 ! 
I've requested some changes, please let me know if something's unclear 👍

**Comment by joaoviana** [comment]
I've also updated the PR description to replace `Closes` with `Relates to` as there's a bit more to do on this ticket 👍

**Comment by joaoviana** [comment]
I'm closing the PR due to inactivity @hrutik7 , please feel free to open again if you wish to continue contributing 🙌  thanks!

**Pull request closed without merging**